### ISSUE TITLE
Find Peak Flow Mods per Second in various scenarios.

### ISSUE THREAD

**Issue by Ktmi** [kytos-ng/flow_manager#193]
This is less an issue, and more of a request for performance measurements. In #187, @viniarck requested I check how pacing affected the flow mod rate in various scenarios. However, we never tested for what the peak flow mod rate would be for flow manager, nor did we test how the max flow mod rate would be affected by latency between the switch and the controller.

**Comment by Ktmi** [comment]
So for my experiment, I decided to try deploying 10,000 flows in a single request via the HTTP api. Here's the IO graph of the packets flow mod packets captured, with a sampling interval of 1 second.
![flow_deployment_max](https://github.com/user-attachments/assets/789bac3e-7249-4eec-b682-cfd6f0d83d44)

This looked to be about 500 is flow mods per second, but didn't really explain the peaks and troughs, so I decided to look at it at a finer sampling interval of 50ms:
![flow_deployment_max_finer](https://github.com/user-attachments/assets/2fb0a225-0461-486f-80c0-f1c90bd31f05)

What we see here is that Kytos is able to sustain sending between 30 and 40 flow mods every 50 ms (or about 600 to 800 flow mods a second), but then has to stop sending flow mods for about a half second.

**Comment by viniarck** [comment]
@Ktmi, that seems similar to my local results, good to have those values documented. I was seding 10k flows for 1 dpid in a single request, the flows mods on wire on a / sec scale, peaked at 1000 (a bit above than what you've found, I'm not running on a VM), and steady with 600 flow mods / sec. So, yes, around 500 flow mods / sec is a good guideline. 

![20240806_141648](https://github.com/user-attachments/assets/6144a2c0-6b45-43d5-827a-4cd48780cd03)

Instrumenting the msg out handler, it also reflected that indeed the number of ops was around 60000 / 60 = 1000 (which matches the peaks at 1000)

![20240806_141507](https://github.com/user-attachments/assets/93dc6047-a7a9-455a-ae1b-57c7d77d76cd)

In a scale of 50 msec, I didn't notice suspensions as you had, but it wasn't steady either

![20240806_141947](https://github.com/user-attachments/assets/5f9aa684-32a3-4b42-afe8-bd9474dc0431)

Let's stick with this 500 flow mods per dpid recommendation. @Ktmi, can you double check if sending for more than one dpid if it still sustains a similar ratio?

**Comment by Ktmi** [comment]
We already have evidence against being able to maintain the same ratio when sending to multiple dpids. In [this post](https://github.com/kytos-ng/flow_manager/pull/187#issuecomment-2253599977), one of the test cases visualized was sending 1000 flow mods to 9 different switches, and the combined flow mod rate was about the same as what I got here, around 500 per second, with the per switch somewhere around 50 per second. It seems that we have a global cap across all DPIDS for how fast we can send flow mods.

**Comment by viniarck** [comment]
> @viniarck We already have evidence against being able to maintain the same ratio when sending to multiple dpids. In [this post](https://github.com/kytos-ng/flow_manager/pull/187#issuecomment-2253599977), one of the test cases visualized was sending 1000 flow mods to 9 different switches, and the combined flow mod rate was about the same as what I got here, around 500 per second, with the per switch somewhere around 50 per second. It seems that we have a global cap across all DPIDS for how fast we can send flow mods.

Right. That's a good data point. OK. Let's keep this in mind, and move on. Let's recommend the 500 flow mods / sec, including globally, and when that's not enough we'll see what else can be done. But, that's already very helpful to to determine certain pacers ceiling. Good.
